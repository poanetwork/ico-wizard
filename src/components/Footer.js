import React from 'react'
import '../assets/stylesheets/application.css';
import { Link } from 'react-router-dom'

export const Footer = () => (
	<footer className="footer">
		<div className="container">
			<p className="rights">2017 Oracles. All rights reserved.</p>
      <Link className="logo" to='/'></Link>
			<div className="socials">
				<button className="social social_reddit"></button>
				<button className="social social_twitter"></button>
				<button className="social social_oracles"></button>
				<button className="social social_telegram"></button>
				<button className="social social_github"></button>
			</div>
		</div>
	</footer>
)