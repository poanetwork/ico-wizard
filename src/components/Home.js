import React, { Component } from 'react';
import '../assets/stylesheets/application.css';
import { Link } from 'react-router-dom'
import { setFlatFileContentToState } from '../utils/utils';
import { defaultState } from '../utils/constants'

export class Home extends Component {
  constructor(props) {
    super(props);
    this.state = defaultState
  }

  addContractsToState (src, bin, abi) {
    let newState = Object.assign({}, this.state)
    newState.contracts.crowdsale = {
      src,
      bin,
      abi: JSON.parse(abi)
    }
    this.setState(newState)
  }

  componentDidMount() {
    const contractName = "SampleCrowdsale";
    let src, bin
    //var contractName = "RomanCrowdsale";
    setFlatFileContentToState("./contracts/" + contractName + "_flat.sol", (content) => src = content);
    setFlatFileContentToState("./contracts/" + contractName + "_flat.bin", (_bin) => bin = _bin);
    setFlatFileContentToState("./contracts/" + contractName + "_flat.abi", (_abi) => this.addContractsToState(src, bin, _abi));
    /*setFlatFileContentToState("./contracts/SampleCrowdsaleToken_flat.bin", function(content) {
      $this.state.contracts.token.bin = content;
    });
    setFlatFileContentToState("./contracts/SampleCrowdsaleToken_flat.abi", function(content) {
      $this.state.contracts.token.abi = JSON.parse(content);
    });*/
  }

  render() {
    return (
      <div>
        <section className="home">
          <div className="crowdsale">
            <div className="container">
              <h1 className="title">Create crowdsale</h1>
              <p className="description">
                Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim
                veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea  commodo consequat.
              </p>
              <div className="buttons">
                <Link to={{ pathname: '/1', query: { state: this.state } }}><a className="button button_fill">New crowdsale</a></Link>
              </div>
            </div>
          </div>
          <div className="process">
            <div className="container">
              <div className="process-item">
                <div className="step-icons step-icons_crowdsale-contract"></div>
                <p className="title">Crowdsale Contract</p>
                <p className="description">
                  Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor
                </p>
              </div>
              <div className="process-item">
                <div className="step-icons step-icons_token-setup"></div>
                <p className="title">Token Setup</p>
                <p className="description">
                  Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor
                </p>
              </div>
              <div className="process-item">
                <div className="step-icons step-icons_crowdsale-setup"></div>
                <p className="title">Crowdsale Setup</p>
                <p className="description">
                  Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor
                </p>
              </div>
              <div className="process-item">
                <div className="step-icons step-icons_publish"></div>
                <p className="title">Publish</p>
                <p className="description">
                  Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor
                </p>
              </div>
              <div className="process-item">
                <div className="step-icons step-icons_crowdsale-page"></div>
                <p className="title">Crowdsale Page</p>
                <p className="description">
                  Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor
                </p>
              </div>
            </div>
          </div>
        </section>
      </div>
    );
  }
}