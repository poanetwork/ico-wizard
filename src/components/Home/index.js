import React, { Component } from 'react';
import '../../assets/stylesheets/application.css';
import { Link } from 'react-router-dom'
import CrowdsalesList from '../Common/CrowdsalesList'
import { Loader } from '../Common/Loader'
import { loadRegistryAddresses } from '../../utils/blockchainHelpers'

export class Home extends Component {
  constructor (props) {
    super(props)
    this.state = {
      showModal: false,
      loading: false
    }
  }

  chooseContract = () => {
    this.setState({
      loading: true
    })

    loadRegistryAddresses()
      .then(() => {
        this.setState({
          loading: false,
          showModal: true
        })
      }, (e) => {
        console.error('There was a problem loading the crowdsale addresses from the registry', e)
        this.setState({
          loading: false
        })
      })
  }

  onClick = crowdsaleAddress => {
    this.props.history.push('/manage/' + crowdsaleAddress)
  }

  renderModal = () => {
    return (
      <div className="crowdsale-modal loading-container">
        <div className='modal'>
          <p className='title'>Crowdsales List</p>
          <p className='description'>Lorem ipsum dolor sit amet, consectetur adipisicing elit. Consectetur cumque dolore
            doloribus ea earumeligendi eveniet ex expedita itaque laborum maxime mollitia nemo nihil perferendis
            quibusdam sed sequiullam, vitae!</p>
          <CrowdsalesList onClick={this.onClick}/>
          <div className='close-button' onClick={() => this.hideModal()}>X</div>
        </div>
      </div>
    )
  }

  hideModal = () => {
    this.setState({ showModal: false })
  }

  render () {
    const chooseContract = process.env.NODE_ENV === 'development'
      ? <div onClick={() => this.chooseContract()} className="button button_outline">Choose Contract</div>
      : null

    const chooseContractModal = this.state.showModal
      ? this.renderModal()
      : null

    return (
      <div>
        <section className="home">
          <div className="crowdsale">
            <div className="container">
              <h1 className="title">Welcome to ICO Wizard</h1>
              <p className="description">
              ICO Wizard is a client side tool to create token and crowdsale contracts in five steps. It helps you to publish contracts on the Ethereum network, verify them in Etherscan, create a crowdsale page with stats. For participants, the wizard creates a page to invest into the campaign.
              <br/>Smart contracts based on <a href="https://github.com/TokenMarketNet/ico">TokenMarket</a> contracts.
              </p>
              <div className="buttons">
                <Link to='/1'><span className="button button_fill">New crowdsale</span></Link>
                {chooseContract}
              </div>
            </div>
          </div>
          <div className="process">
            <div className="container">
              <div className="process-item">
                <div className="step-icons step-icons_crowdsale-contract"></div>
                <p className="title">Crowdsale Contract</p>
                <p className="description">
                  Select a strategy for crowdsale contract
                </p>
              </div>
              <div className="process-item">
                <div className="step-icons step-icons_token-setup"></div>
                <p className="title">Token Setup</p>
                <p className="description">
                  Setup token and reserved distribution
                </p>
              </div>
              <div className="process-item">
                <div className="step-icons step-icons_crowdsale-setup"></div>
                <p className="title">Crowdsale Setup</p>
                <p className="description">
                  Setup tiers and crowdsale parameters
                </p>
              </div>
              <div className="process-item">
                <div className="step-icons step-icons_publish"></div>
                <p className="title">Publish</p>
                <p className="description">
                  Get generated code and artifacts for verification in Etherscan
                </p>
              </div>
              <div className="process-item">
                <div className="step-icons step-icons_crowdsale-page"></div>
                <p className="title">Crowdsale Page</p>
                <p className="description">
                  Bookmark this page for the campaign statistics
                </p>
              </div>
            </div>
          </div>
          {chooseContractModal}
          <Loader show={this.state.loading}></Loader>
        </section>
      </div>
    );
  }
}
