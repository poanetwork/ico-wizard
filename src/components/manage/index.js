import React, { Component } from 'react'
import { inject, observer } from 'mobx-react'
import { Link } from 'react-router-dom'
import { CONTRACT_TYPES, TEXT_FIELDS, TOAST, VALIDATION_MESSAGES } from '../../utils/constants'
import { InputField } from '../Common/InputField'
import '../../assets/stylesheets/application.css'
import { WhitelistInputBlock } from '../Common/WhitelistInputBlock'
import { successfulFinalizeAlert, warningOnFinalizeCrowdsale } from '../../utils/alerts'
import { getNetworkVersion, sendTXToContract } from '../../utils/blockchainHelpers'
import { getWhiteListWithCapCrowdsaleAssets, toast } from '../../utils/utils'
import { findCurrentContractRecursively, getCurrentRate } from '../crowdsale/utils'
import { getTiers, processTier, updateTierAttribute } from './utils'
import { Loader } from '../Common/Loader'

const { START_TIME, END_TIME, RATE, SUPPLY, WALLET_ADDRESS, CROWDSALE_SETUP_NAME } = TEXT_FIELDS

@inject('crowdsaleStore', 'web3Store', 'tierStore', 'contractStore', 'crowdsalePageStore', 'generalStore', 'tokenStore')

@observer
export class Manage extends Component {
  constructor (props) {
    super(props)
    window.scrollTo(0, 0)
    this.state = {
      formPristine: true,
      loading: true
    }
  }

  componentWillMount () {
    const { crowdsaleStore, web3Store, contractStore, generalStore, match } = this.props
    const crowdsaleAddress = match.params.crowdsaleAddress

    crowdsaleStore.setSelectedProperty('address', crowdsaleAddress)

    // networkID
    getNetworkVersion(web3Store.web3).then(networkId => generalStore.setProperty('networkId', networkId))

    // contractType
    contractStore.setContractType(CONTRACT_TYPES.whitelistwithcap)

    getWhiteListWithCapCrowdsaleAssets().then(this.extractContractData)
  }

  extractContractData = () => {
    const { contractStore, crowdsaleStore } = this.props

    getTiers(crowdsaleStore.selected.address)
      .then(joinedCrowdsales => {
        contractStore.setContractProperty('crowdsale', 'addr', joinedCrowdsales)

        if (!contractStore.crowdsale.addr) {
          this.hideLoader()
          return Promise.reject('no tiers addresses')
        }
      })
      .then(() => {
        return contractStore.crowdsale.addr.reduce((promise, addr, index) => {
          return promise.then(() => processTier(addr, index))
        }, Promise.resolve())
      })
      .catch(console.log)
      .then(this.hideLoader)
  }

  hideLoader = () => {
    this.setState({ loading: false })
  }

  showLoader = () => {
    this.setState({ loading: true })
  }

  isLastContract = crowdsaleContract => {
    const { contractStore } = this.props
    const crowdsalesAddresses = contractStore.crowdsale.addr
    return crowdsalesAddresses[crowdsalesAddresses.length - 1] === crowdsaleContract._address
  }

  finalizeCrowdsale = () => {
    const { crowdsaleStore, web3Store } = this.props
    const { web3 } = web3Store

    if (!crowdsaleStore.selected.finalized) {
      warningOnFinalizeCrowdsale()
        .then(result => {
          if (result.value) {
            findCurrentContractRecursively(0, this, web3, null, crowdsaleContract => {
              this.showLoader()

              if (!this.isLastContract(crowdsaleContract)) {
                this.hideLoader()
                toast.showToaster({ type: TOAST.TYPE.ERROR, message: TOAST.MESSAGE.FINALIZE_FAIL })

              } else {
                const finalizeMethod = crowdsaleContract.methods.finalize().send({
                  gasLimit: 650000,
                  gasPrice: this.props.generalStore.gasPrice
                })

                getCurrentRate(web3, crowdsaleContract)
                  .then(() => sendTXToContract(web3, finalizeMethod))
                  .then(() => {
                    successfulFinalizeAlert()
                    crowdsaleStore.setSelectedProperty('finalized', true)
                  })
                  .catch(err => toast.showToaster({ type: TOAST.TYPE.ERROR, message: TOAST.MESSAGE.FINALIZE_FAIL }))
                  .then(this.hideLoader)
              }
            })

          }
        })
    }
  }

  saveCrowdsale = e => {
    e.preventDefault()
    e.stopPropagation()

    if (!this.state.formPristine) {
      const { crowdsaleStore, tierStore } = this.props
      const updatableTiers = crowdsaleStore.selected.initialTiersValues.filter(tier => tier.updatable)

      if (updatableTiers.length) {
        const isValidTier = tierStore.individuallyValidTiers
        console.log(isValidTier)

        const validTiers = updatableTiers.every(tier => isValidTier[tier.index])

        if (validTiers) {
          const keys = Object.keys(updatableTiers[0])
            .filter(key => key !== 'index' && key !== 'updatable' && key !== 'addresses')

          const attributesToUpdate = updatableTiers.reduce((toUpdate, tier) => {
            keys.forEach(key => {
              const newValue = tierStore.tiers[tier.index][key]
              const { addresses } = tier

              if (newValue !== tier[key]) {
                toUpdate.push({ key, newValue, addresses })
              }
            })

            return toUpdate
          }, [])

          this.showLoader()

          attributesToUpdate.reduce((promise, { key, newValue, addresses }) => {
            return promise.then(() => updateTierAttribute(key, newValue, addresses))
          }, Promise.resolve())
            .catch(err => {
              console.log(err)
              toast.showToaster({ type: TOAST.TYPE.ERROR, message: TOAST.MESSAGE.TRANSACTION_FAILED })
            })
            .then(this.hideLoader)
        }
      }
    }
  }

  changeState = (event, parent, key, property) => {
    const { tierStore } = this.props
    const whitelistInputProps = { ...tierStore.tiers[key].whitelistInput }
    const prop = property.split('_')[1]
    whitelistInputProps[prop] = event.target.value
    tierStore.setTierProperty(whitelistInputProps, 'whitelistInput', key)
  }

  clickedWhiteListInputBlock = e => {
    if (e.target.classList.contains('button_fill_plus')) {
      this.setState({ formPristine: false })
    }
  }

  whitelistInputBlock = index => {
    return (
      <WhitelistInputBlock
        key={index.toString()}
        num={index}
        onChange={(e, contract, num, prop) => this.changeState(e, contract, num, prop)}
      />
    )
  }

  readOnlyWhitelistedAddresses = tier => {
    if (!tier.whitelist.length) {
      return (
        <div className="white-list-item-container">
          <div className="white-list-item-container-inner">
            <span className="white-list-item white-list-item-left">no addresses loaded</span>
          </div>
        </div>
      )
    }

    return tier.whitelist.map(item => (
      <div className={'white-list-item-container'} key={item.addr}>
        <div className="white-list-item-container-inner">
          <span className="white-list-item white-list-item-left">{item.addr}</span>
          <span className="white-list-item white-list-item-middle">{item.min}</span>
          <span className="white-list-item white-list-item-right">{item.max}</span>
        </div>
      </div>
    ))
  }

  renderWhitelistInputBlock = (tier, index) => {
    const { crowdsaleStore, tierStore } = this.props

    if (tierStore.tiers[0].whitelistdisabled === 'yes') {
      return null
    }

    return (
      <div onClick={this.clickedWhiteListInputBlock}>
        <div className="section-title">
          <p className="title">Whitelist</p>
        </div>
        {tier.updatable && !crowdsaleStore.selected.finalized
          ? this.whitelistInputBlock(index)
          : this.readOnlyWhitelistedAddresses(tier)
        }
      </div>
    )
  }

  updateTierStore = (event, property, index) => {
    const { tierStore } = this.props
    const value = event.target.value

    tierStore.setTierProperty(value, property, index)

    if (property === 'endTime' || property === 'startTime') {
      tierStore.validateEditedTier(property, index)
    } else {
      tierStore.validateTiers(property, index)
    }

    if (this.state.formPristine) {
      this.setState({ formPristine: false })
    }
  }

  render () {
    const { formPristine } = this.state
    const { generalStore, tierStore, tokenStore, crowdsaleStore } = this.props
    const { address: crowdsaleAddress, finalized, updatable } = crowdsaleStore.selected

    const aboutStep = (
      <div className="about-step">
        <div className="swal2-icon swal2-info warning-logo">!</div>
        <p className="title">Finalize Crowdsale</p>
        <p className="description">Finalize - Finalization is the last step of the crowdsale.
          You can make it only after the end of the last tier. After finalization, it's not possible to update
          tiers, buy tokens. All tokens will be movable, reserved tokens will be issued.</p>
        <Link to='#' onClick={() => this.finalizeCrowdsale()}>
          <span className={`button button_${finalized ? 'disabled' : 'fill'}`}>Finalize Crowdsale</span>
        </Link>
      </div>
    )

    const aboutTier = (
      <div className="about-step">
        <div className="step-icons step-icons_crowdsale-setup"/>
        <p className="title">{tokenStore.name} ({tokenStore.ticker}) Settings</p>
        <p className="description">The most important and exciting part of the crowdsale
          process. Here you can define parameters of your crowdsale campaign.</p>
        <Link to={`/crowdsale/?addr=${crowdsaleAddress}&networkID=${generalStore.networkId}`}
              className="crowdsale-page-link"
        >Crowdsale page</Link>
      </div>
    )

    const saveButton = (
      <Link to='/2' onClick={e => this.saveCrowdsale(e)}>
        <span className={`no-arrow button button_${!formPristine ? 'fill' : 'disabled'}`}>Save</span>
      </Link>
    )

    const tierNameAndWallet = (tier) => {
      return <div className='input-block-container'>
        <InputField
          side='left'
          type='text'
          title={CROWDSALE_SETUP_NAME}
          value={tier.name}
          disabled
        />
        <InputField
          side='right'
          type='text'
          title={WALLET_ADDRESS}
          value={tier.walletAddress}
          disabled
        />
      </div>
    }

    const tierStartAndEndTime = (tier, index) => {
      return <div className='input-block-container'>
        <InputField
          side='left'
          type='datetime-local'
          title={START_TIME}
          value={tier.startTime}
          valid={tierStore.validTiers[index] && tierStore.validTiers[index].startTime}
          errorMessage={VALIDATION_MESSAGES.EDITED_START_TIME}
          onChange={(e) => this.updateTierStore(e, 'startTime', index)}
          description="Date and time when the tier starts. Can't be in the past from the current moment."
          disabled={!tier.updatable || finalized}
        />
        <InputField
          side='right'
          type='datetime-local'
          title={END_TIME}
          value={tier.endTime}
          valid={tierStore.validTiers[index] && tierStore.validTiers[index].endTime}
          errorMessage={VALIDATION_MESSAGES.EDITED_END_TIME}
          onChange={(e) => this.updateTierStore(e, 'endTime', index)}
          description="Date and time when the tier ends. Can be only in the future."
          disabled={!tier.updatable || finalized}
        />
      </div>
    }

    const tierRateAndSupply = (tier, index) => {
      return <div className='input-block-container'>
        <InputField
          side='left'
          type='number'
          title={RATE}
          value={tier.rate}
          valid={tierStore.validTiers[index] && tierStore.validTiers[index].rate}
          errorMessage={VALIDATION_MESSAGES.RATE}
          onChange={(e) => this.updateTierStore(e, 'rate', index)}
          description="Exchange rate Ethereum to Tokens. If it's 100, then for 1 Ether you can buy 100 tokens"
          disabled={!tier.updatable || finalized}
        />
        <InputField
          side='right'
          type='number'
          title={SUPPLY}
          value={tier.supply}
          valid={tierStore.validTiers[index] && tierStore.validTiers[index].supply}
          errorMessage={VALIDATION_MESSAGES.SUPPLY}
          onChange={(e) => this.updateTierStore(e, 'supply', index)}
          description="How many tokens will be sold on this tier. Cap of crowdsale equals to sum of supply of all tiers"
          disabled={!tier.updatable || finalized}
        />
      </div>
    }

    return (
      <section className="manage">
        <div className="steps-content container">
          {aboutStep}
        </div>
        {tierStore.tiers.map((tier, index) => (
          <div className="steps" key={index.toString()}>
            <div className="steps-content container">
              {index === 0 ? aboutTier : null}
              <div className={`hidden ${tierStore.tiers[0].whitelistdisabled !== 'yes' ? 'divisor' : ''}`}>
                {tierNameAndWallet(tier)}
                {tierStartAndEndTime(tier, index)}
                {tierRateAndSupply(tier, index)}
              </div>
              {this.renderWhitelistInputBlock(tier, index)}
            </div>
          </div>
        ))}
        <div className="steps">
          <div className="button-container">
            {!finalized && updatable ? saveButton : null}
          </div>
        </div>
        <Loader show={this.state.loading}/>
      </section>
    )
  }
}
