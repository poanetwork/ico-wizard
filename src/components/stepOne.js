import React from 'react'
import '../assets/stylesheets/application.css';
import { Link } from 'react-router-dom'
import { defaultState } from '../utils/constants'
import { getOldState } from '../utils/utils'

export class stepOne extends React.Component {
  constructor(props) {
    super(props);
    let oldState = getOldState(props, defaultState)
    this.state = Object.assign({}, oldState)
  }

  render() {
    return (
    	 <section className="steps steps_crowdsale-contract">
        <div className="steps-navigation">
          <div className="container">
            <div className="step-navigation step-navigation_active">Crowdsale Contract</div>
            <div className="step-navigation">Token Setup</div>
            <div className="step-navigation">Crowdsale Setup</div>
            <div className="step-navigation">Publish</div>
            <div className="step-navigation">Crowdsale Page</div>
          </div>
        </div>
        <div className="steps-content container">
          <div className="about-step">
            <div className="step-icons step-icons_crowdsale-contract"></div>
            <p className="title">Crowdsale Contract</p>
            <p className="description">
              Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut
              enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor
              in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur.
            </p>
          </div>
          <div className="radios">
            <label className="radio">
              <input type="radio" checked name="contract-type"/>
              <span className="title">Standard</span>
              <span className="description">
                Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam,
                quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.
              </span>
            </label>
            <label className="radio">
              <input type="radio" disabled name="contract-type"/>
              <span className="title title_soon">Capped</span>
              <span className="description">
                Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam,
                quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.
              </span>
            </label>
          </div>
        </div>
        <div className="button-container">
          <Link to={{ pathname: '/2', query: { state: this.state } }}><a className="button button_fill">Continue</a></Link>
        </div>
      </section>
    )}
}