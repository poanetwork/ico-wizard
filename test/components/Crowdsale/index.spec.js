import React from 'react'
import Adapter from 'enzyme-adapter-react-15'
import { Provider } from 'mobx-react'
import { configure } from 'enzyme'
import renderer from 'react-test-renderer'
import { MemoryRouter } from 'react-router'
import {
  contractStore,
  crowdsaleStore,
  crowdsalePageStore,
  web3Store,
  tierStore,
  tokenStore,
  generalStore
} from '../../../src/stores'
import { Crowdsale } from '../../../src/components/Crowdsale'

configure({ adapter: new Adapter() })

describe('Crowdsale index', () => {
  const stores = { contractStore, crowdsaleStore, crowdsalePageStore, web3Store, tierStore, tokenStore, generalStore }

  it(`should render Crowdsale`, () => {
    // Given
    const component = renderer.create(
      <Provider {...stores}>
        <MemoryRouter initialEntries={['/crowdsale']}>
          <Crowdsale />
        </MemoryRouter>
      </Provider>
    )

    // When
    const tree = component.toJSON()

    // Then
    expect(tree).toMatchSnapshot()
  })
})
