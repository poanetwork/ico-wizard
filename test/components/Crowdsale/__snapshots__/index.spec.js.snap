// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Crowdsale index should render Crowdsale 1`] = `
<div>
  <section
    className="lo-MenuBarAndContent"
  >
    <div
      className="st-StepNavigation"
    >
      <a
        className="sw-LogoWhite"
        href="/"
      />
      <div
        className="st-StepNavigation_Container"
      >
        <div
          className="st-StepItem
        active  "
        >
          <div
            className="st-StepItem_Circle"
          />
          <div
            className="st-StepItem_Line"
          />
          <span
            className="st-StepItem_Text"
          >
            Crowdsale Strategy
          </span>
        </div>
        <div
          className="st-StepItem
        active  "
        >
          <div
            className="st-StepItem_Circle"
          />
          <div
            className="st-StepItem_Line"
          />
          <span
            className="st-StepItem_Text"
          >
            Token Setup
          </span>
        </div>
        <div
          className="st-StepItem
        active  "
        >
          <div
            className="st-StepItem_Circle"
          />
          <div
            className="st-StepItem_Line"
          />
          <span
            className="st-StepItem_Text"
          >
            Crowdsale Setup
          </span>
        </div>
        <div
          className="st-StepItem
        active  "
        >
          <div
            className="st-StepItem_Circle"
          />
          <div
            className="st-StepItem_Line"
          />
          <span
            className="st-StepItem_Text"
          >
            Publish
          </span>
        </div>
        <div
          className="st-StepItem
        active  "
        >
          <div
            className="st-StepItem_Circle"
          />
          <span
            className="st-StepItem_Text"
          >
            Crowdsale Page
          </span>
        </div>
      </div>
    </div>
    <div
      className="st-StepContent"
    >
      <div
        className="st-SectionInfo "
      >
        <div
          className="st-SectionInfo_Icon st-SectionInfo_Icon-step5"
        />
        <div
          className="st-SectionInfo_Text"
        >
          <h1
            className="st-SectionInfo_Title"
          >
            Crowdsale Page
          </h1>
          <p
            className="st-SectionInfo_Description"
            dangerouslySetInnerHTML={
              Object {
                "__html": "Page with statistics of crowdsale. Statistics for all tiers combined on the page. Please press Ctrl-D to bookmark the page.",
              }
            }
          />
        </div>
      </div>
      <div
        className="cs-CrowdsaleProgress "
      >
        <div
          className="cs-CrowdsaleProgress_Funds"
        >
          <div
            className="cs-CrowdsaleProgress_FundsAmount cs-CrowdsaleProgress_FundsAmount-total-raised"
          >
            <h2
              className="cs-CrowdsaleProgress_FundsTitle"
            >
              0
               
              <span
                className="cs-CrowdsaleProgress_NoBreak"
              >
                ETH
              </span>
            </h2>
            <p
              className="cs-CrowdsaleProgress_FundsDescription"
            >
              Total Raised Funds
            </p>
          </div>
          <div
            className="cs-CrowdsaleProgress_FundsAmount cs-CrowdsaleProgress_FundsAmount-goal"
          >
            <h2
              className="cs-CrowdsaleProgress_FundsTitle"
            >
              0
               
              <span
                className="cs-CrowdsaleProgress_NoBreak"
              >
                ETH
              </span>
            </h2>
            <p
              className="cs-CrowdsaleProgress_FundsDescription"
            >
              Goal
            </p>
          </div>
        </div>
        <div
          className="cs-CrowdsaleProgress_Bar"
        >
          <div
            className="cs-CrowdsaleProgress_BarWidth"
            style={
              Object {
                "width": "0%",
              }
            }
          />
          <div
            className="cs-CrowdsaleProgress_Division"
          />
          <div
            className="cs-CrowdsaleProgress_Division"
          />
          <div
            className="cs-CrowdsaleProgress_Division"
          />
          <div
            className="cs-CrowdsaleProgress_Division"
          />
          <div
            className="cs-CrowdsaleProgress_Division"
          />
          <div
            className="cs-CrowdsaleProgress_Division"
          />
          <div
            className="cs-CrowdsaleProgress_Division"
          />
          <div
            className="cs-CrowdsaleProgress_Division"
          />
          <div
            className="cs-CrowdsaleProgress_Division"
          />
        </div>
      </div>
      <div
        className="cs-CrowdsaleID "
      >
        <div
          className="cs-CrowdsaleID_InnerContainer"
        >
          <h3
            className="cs-CrowdsaleID_Hash"
          >
            <span
              className="cs-CrowdsaleID_HashText"
            />
             
            <button
              className="sw-ButtonCopyToClipboard "
              data-clipboard-action="copy"
              onClick={[Function]}
              type="button"
            >
              <svg
                height="20"
                viewBox="0 0 16 20"
                width="16"
                xmlns="http://www.w3.org/2000/svg"
              >
                <g
                  fill="#8CE1D7"
                  fillRule="nonzero"
                >
                  <path
                    d="M10.867 3.493H1.602c-.87 0-1.575.707-1.575 1.579v13.32c0 .872.706 1.58 1.575 1.58h9.265c.87 0 1.576-.708 1.576-1.58V5.073a1.583 1.583 0 0 0-1.576-1.58zm.47 14.896a.474.474 0 0 1-.474.474H1.598a.474.474 0 0 1-.474-.474V5.072c0-.262.213-.475.474-.475h9.265c.261 0 .474.213.474.475v13.317z"
                  />
                  <path
                    d="M14.41 0H5.145c-.87 0-1.576.708-1.576 1.579a.55.55 0 0 0 .551.552.55.55 0 0 0 .551-.552c0-.262.213-.475.474-.475h9.265c.261 0 .474.213.474.475v13.32a.474.474 0 0 1-.474.475.55.55 0 0 0-.55.552.55.55 0 0 0 .55.553c.87 0 1.576-.708 1.576-1.58V1.58C15.986.708 15.28 0 14.41 0z"
                  />
                </g>
              </svg>
            </button>
          </h3>
          <p
            className="cs-CrowdsaleID_Description"
          >
            Crowdsale Proxy Address
          </p>
        </div>
      </div>
      <div
        className="st-StepContent_Buttons"
      >
        <button
          className="sw-ButtonContinue "
          disabled={false}
          onClick={[Function]}
          type="button"
        >
          <span
            className="sw-ButtonContinue_Text"
          >
            Contribute
          </span>
        </button>
      </div>
    </div>
  </section>
  <div
    className="sw-FullscreenBackdrop"
  >
    <div
      className="sw-Loader"
    >
      <div
        className="sw-Loader_AnimatedItem"
      />
      <div
        className="sw-Loader_AnimatedItem"
      />
      <div
        className="sw-Loader_AnimatedItem"
      />
      <div
        className="sw-Loader_AnimatedItem"
      />
      <div
        className="sw-Loader_AnimatedItem"
      />
      <div
        className="sw-Loader_AnimatedItem"
      />
    </div>
  </div>
</div>
`;
